runtime/opt/taupage/init.d/01-register-scalyr-agent.py

```diff
@@ -0,0 +1,214 @@
+        'replacement': '+++JWT_TOKEN_REDACTED+++'
+    }
+]
+
+if rsyslog_aws_metadata:
+    sys_log_parser = 'systemLogMetadata'
+else:
+    sys_log_parser = 'systemLog'
+
+if not logging_config:
+    account_key = main_config.get('scalyr_account_key')
+    region = main_config.get('scalyr_region', 'eu')
+    application_log_parser = main_config.get('scalyr_application_log_parser', 'slf4j')
+    custom_log_parser = main_config.get('scalyr_application_log_parser', 'slf4j')
+    scalyr_agent_enabled = True
+    ship_application_log = True
+    application_log_sampling = None
+    ship_custom_log = True
+    custom_log_sampling = None
+    ship_auth_log = True
+    auth_log_sampling = None
+    ship_sys_log = True
+    sys_log_sampling = None
+else:
+    account_key = logging_config.get('scalyr_account_key')
+    region = logging_config.get('scalyr_region', 'eu')
+    application_log_parser = logging_config.get('scalyr_application_log_parser', 'slf4j')
+    custom_log_parser = logging_config.get('scalyr_application_log_parser', 'slf4j')
+    fluentd_enabled = logging_config.get('fluentd_enabled')
+    scalyr_agent_enabled = logging_config.get('scalyr_agent_enabled', not fluentd_enabled)
+    ship_all_logs = logging_config.get('use_scalyr_agent_all', not fluentd_enabled)
+    ship_application_log = logging_config.get('use_scalyr_agent_applog', ship_all_logs)
+    application_log_sampling = logging_config.get('scalyr_agent_applog_sampling')
+    ship_custom_log = logging_config.get('use_scalyr_agent_customlog', ship_all_logs)
+    custom_log_sampling = logging_config.get('scalyr_agent_customlog_sampling')
+    ship_auth_log = logging_config.get('use_scalyr_agent_authlog', ship_all_logs)
+    auth_log_sampling = logging_config.get('scalyr_agent_authlog_sampling')
+    ship_sys_log = logging_config.get('use_scalyr_agent_syslog', ship_all_logs)
+    sys_log_sampling = logging_config.get('scalyr_agent_syslog_sampling')
+
+if not (scalyr_agent_enabled and account_key):
+    logger.info('Found no Scalyr key or Scalyr agent not enabled; skipping Scalyr Agent init')
+    raise SystemExit()
+
+if not (ship_application_log or ship_custom_log or ship_auth_log or ship_sys_log):
+    logger.info('Nothing to do for Scalyr Agent; skipping Scalyr Agent init')
+    raise SystemExit()
+
+
+def restart_scalyr_agent_process():
+    ''' Restart SScalyr Agent '''
+    if account_key == "foo1234":
+        return 0
+    process = subprocess.Popen(['/etc/init.d/scalyr-agent-2', 'restart'])
+    exit_code = process.wait(timeout=5)
+    if exit_code:
+        raise Exception('"/etc/init.d/scalyr-agent-2 restart" failed with exit code: {0}'.format(exit_code))
+
+
+def decrypt_scalyr_key():
+    scalyr_api_key = None
+    match_kms_key = re.search('aws:kms:', account_key, re.IGNORECASE)
+    if match_kms_key:
+        scalyr_api_key = re.sub(r'aws:kms:', '', account_key)
+        try:
+            scalyr_api_key = subprocess.check_output(['python3',
```

Comment:
I'd move KMS decryption to the `taupage` package and just call it directly. Shelling out from Python to Python and then checking for success based on a random string in `stdout` could be avoided.

Reply:
Alternatively, just call the KMS client directly, it's just 2-3 lines of code.